react/introduction/setting_up_a_react_environment.md

```diff
@@ -0,0 +1,145 @@
+
+### Lesson Overview
+
+This section contains a general overview of topics that you will learn in this lesson.
+
+* How React projects can be created locally.
+* Describing the difficulties when starting a React project without the appropriate tools.
+* Listing some of the popular tools and methods for scaffolding a new React project.
+* Explaining what Create React App is.
+* Setting up your workflow to develop efficiently within your new project.
+
+### Many Paths
+
+There are multiple ways to start using React in your projects, from attaching a set of `<script>` tags which serve React from a [CDN](https://en.wikipedia.org/wiki/Content_delivery_network), to robust [toolchains](https://en.wikipedia.org/wiki/Toolchain) that are highly configurable and allow for increased scalability and optimization.
+
+Some examples of these toolchains:
+
+* [Create React App](https://create-react-app.dev/)
+* [Vite's React Config](https://vitejs.dev/)
+* [Gatsby](https://www.gatsbyjs.com/)
+* [NextJS](https://nextjs.org/)
+
+Why do we need these toolchains? Can't we just make our own as we see fit?
+
+Yes, but it's _hard_. React is a complex beast and there are many moving parts. Before you can start writing any sort of code that provides functionality, you would need to configure _at least_ the following:
+
+* Package Management (NPM, Yarn)
+* Module bundling (Webpack, Parcel)
+* Compilation (Babel)
+* React itself
+
+All of this, and sometimes _much more_ is required to get a React project and development environment up and running.
+
+### Simplifying The Process
+
+Now that you understand what is involved with starting a React project from scratch, you can breathe a sigh of relief as you learn that we can get started with a _single terminal command_.
+
+This terminal command is a feature of the beginner-friendly [Create React App](https://create-react-app.dev/), a tool developed by the folks over at Meta which we will be using from here on out. It requires minimal configuration and provides extremely useful tools straight out of the box, allowing us to get straight to the learning. Let's get started!
+
+### Creating A React App
+
+Fire up a terminal instance, `cd` over to the folder containing your projects and enter the following command:
+
+~~~
+$ npx create-react-app my-first-react-app
+~~~
+
+Note that we use `npx` here (not `npm`), this is not a typo, `npx` is a package runner that allows the execution of packages without them being installed locally on your machine. [Check here](https://www.geeksforgeeks.org/what-are-the-differences-between-npm-and-npx/) for more information.
+
+Once this process has completed, we should confirm that all has gone well:
+
+~~~
+$ cd my-first-react-app
+$ npm start
+~~~
+
+Provided that everything has gone according to plan, you should be greeted with a new browser tab containing a lovely spinning React logo:
+
+![Create React App Starting Page](./setting_up_a_react_environment/imgs/cra_start.png)
```

Comment:
Image link will need to be replaced by a statically link ([using this guide](https://github.com/TheOdinProject/curriculum/wiki/Adding-Images-to-the-Curriculum)) after the lesson is merged.